lib/pbio/src/color/util.c

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022 The Pybricks Authors
+
+#include <pbio/color.h>
+
+int32_t sin_deg_branch0(int32_t x) {
+    return 201 * x - x * x;
+}
+
+// integer sine approximation from degrees to (-10000, 10000)
+int32_t sin_deg(int32_t x) {
+    x = x % 360;
+    if (x < 90) {
+        return sin_deg_branch0(x);
+    }
+    if (x < 180) {
+        return sin_deg_branch0(180 - x);
+    }
+    if (x < 270) {
+        return -sin_deg_branch0(x - 180);
+    }
+    return -sin_deg_branch0(360 - x);
+}
+
+int32_t cos_deg(int32_t x) {
+    return sin_deg(x + 90);
+}
+
+// gets squared cartesian distance between hsv colors mapped into a chroma-lightness-bicone
```

Comment:
It would be nice to have doxygen style comments for public functions. And to clarify what the units of the return value are.

```suggestion
/**
 * Gets squared Cartesian distance between HSV colors mapped into a chroma-lightness-bicone.
 * @param [in]  hsv_a    The first HSV color.
 * @param [in]  hsv_b    The second HSV color.
 * @returns              This distance (0 to ???).
 */
```